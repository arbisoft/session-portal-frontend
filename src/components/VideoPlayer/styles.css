@import "@vidstack/react/player/styles/default/theme.css";
@import "@vidstack/react/player/styles/default/layouts/audio.css";
@import "@vidstack/react/player/styles/default/layouts/video.css";

.player {
  --brand-color: #f5f5f5;
  --focus-color: #4e9cf6;

  --audio-brand: var(--brand-color);
  --audio-focus-ring-color: var(--focus-color);
  --audio-border-radius: 2px;

  --video-brand: var(--brand-color);
  --video-focus-ring-color: var(--focus-color);
  --video-border-radius: 2px;

  /* ðŸ‘‰ https://vidstack.io/docs/player/components/layouts/default#css-variables for more. */

  [data-view-type="video"] {
    aspect-ratio: 16 /9;
  }

  .vds-google-cast-button{
    display: none;
  }

  .vds-main-play-button {
    background: #eeeeee1b;
    border: 0;
    bottom: 0;
    color: rgba(0, 0, 0, 0.30);
    cursor: pointer;
    left: 0;
    margin: 0;
    outline: none;
    padding: 0;
    position: absolute;
    right: 0;
    top: 0;
    z-index: 1;

    .play-icon {
      border-radius: 50%;
      box-shadow: 0px 0px 20px 0px rgba(0, 0, 0, 0.30);
      font-size: 150px;

      @media (max-width: 1199.95px) {
        font-size: 80px;
      }
    }
  }
}
