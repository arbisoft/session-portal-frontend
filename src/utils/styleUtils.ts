export const shouldForwardProp = (propName: string) => !propName.startsWith("$");

export const pxToRem = (px: number, base = 16): string => {
  return `${(px * 1.25) / base}rem`;
};
