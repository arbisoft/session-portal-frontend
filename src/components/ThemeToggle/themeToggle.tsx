import React from "react";

import Brightness4 from "@mui/icons-material/Brightness4";
import Brightness7 from "@mui/icons-material/Brightness7";
import IconButton from "@mui/material/IconButton";
import { useColorScheme, alpha, useTheme } from "@mui/material/styles";
import Tooltip from "@mui/material/Tooltip";

const ThemeToggle: React.FC = () => {
  const theme = useTheme();
  const { mode, setMode } = useColorScheme();

  return (
    <Tooltip title={mode === "dark" ? "Switch to light mode" : "Switch to dark mode"}>
      <IconButton
        role="dark-mode-button"
        onClick={() => setMode(mode === "dark" ? "light" : "dark")}
        size="medium"
        color="inherit"
        sx={{
          ml: 1,
          bgcolor: mode === "dark" ? alpha(theme.palette.common.white, 0.08) : alpha(theme.palette.common.black, 0.04),
          "&:hover": {
            bgcolor: mode === "dark" ? alpha(theme.palette.common.white, 0.12) : alpha(theme.palette.common.black, 0.08),
          },
        }}
      >
        {mode === "dark" ? <Brightness7 /> : <Brightness4 />}
      </IconButton>
    </Tooltip>
  );
};

export default ThemeToggle;
