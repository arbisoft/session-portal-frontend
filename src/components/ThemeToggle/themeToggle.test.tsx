import { useColorScheme } from "@mui/material/styles";

import { customRender as render, screen, fireEvent } from "@/jest/utils/testUtils";

import ThemeToggle from "./themeToggle";

// Mock useColorScheme
jest.mock("@mui/material/styles", () => ({
  ...jest.requireActual("@mui/material/styles"),
  useColorScheme: jest.fn(),
}));

describe("ThemeToggle", () => {
  const mockSetMode = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders light mode icon when current mode is dark", () => {
    (useColorScheme as jest.Mock).mockReturnValue({
      mode: "dark",
      setMode: mockSetMode,
    });

    render(<ThemeToggle />);

    // Check that the light mode icon is rendered (Brightness7)
    expect(screen.getByRole("dark-mode-button")).toBeInTheDocument();
    expect(screen.getByTestId("Brightness7Icon")).toBeInTheDocument();
  });

  it("renders dark mode icon when current mode is light", () => {
    (useColorScheme as jest.Mock).mockReturnValue({
      mode: "light",
      setMode: mockSetMode,
    });

    render(<ThemeToggle />);

    // Check that the dark mode icon is rendered (Brightness4)
    expect(screen.getByRole("dark-mode-button")).toBeInTheDocument();
    expect(screen.getByTestId("Brightness4Icon")).toBeInTheDocument();
  });

  it("toggles from dark to light mode on click", () => {
    (useColorScheme as jest.Mock).mockReturnValue({
      mode: "dark",
      setMode: mockSetMode,
    });

    render(<ThemeToggle />);
    const toggleButton = screen.getByRole("dark-mode-button");

    fireEvent.click(toggleButton);

    expect(mockSetMode).toHaveBeenCalledWith("light");
  });

  it("toggles from light to dark mode on click", () => {
    (useColorScheme as jest.Mock).mockReturnValue({
      mode: "light",
      setMode: mockSetMode,
    });

    render(<ThemeToggle />);
    const toggleButton = screen.getByRole("dark-mode-button");

    fireEvent.click(toggleButton);

    expect(mockSetMode).toHaveBeenCalledWith("dark");
  });

  it("matches snapshot", () => {
    (useColorScheme as jest.Mock).mockReturnValue({
      mode: "light",
      setMode: mockSetMode,
    });

    const { asFragment } = render(<ThemeToggle />);
    expect(asFragment()).toMatchSnapshot();
  });
});
