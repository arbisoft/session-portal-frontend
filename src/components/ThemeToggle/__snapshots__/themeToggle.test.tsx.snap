// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggle matches snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Switch to dark mode"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-hcxzx-MuiButtonBase-root-MuiIconButton-root"
    data-mui-internal-clone-element="true"
    role="dark-mode-button"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4uxqju-MuiSvgIcon-root"
      data-testid="Brightness4Icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6"
      />
    </svg>
  </button>
</DocumentFragment>
`;
